src/IRsend.cpp

```diff
@@ -20,10 +20,27 @@
 // sending IR code on ESP8266
 
 // IRsend ----------------------------------------------------------------------
-
-IRsend::IRsend(uint16_t IRsendPin) {
+// Create an IRsend object.
+//
+// Args:
+//   IRsendPin:  Which GPIO pin to use when sending an IR command.
+//   inverted:   *DANGER* Optional flag to invert the output. (default = false)
+//               e.g. LED is illuminated when GPIO is LOW rather than HIGH.
+//               Setting this to something other than the default could
+//               easily destroy your IR LED if you are overdriving it.
+//               Unless you *REALLY* know what you are doing, don't change this.
+// Returns:
+//   An IRsend object.
+IRsend::IRsend(uint16_t IRsendPin, bool inverted) {
   IRpin = IRsendPin;
   periodOffset = PERIOD_OFFSET;
+  if (inverted) {
+    outputOn = LOW;
```

Comment:
For the record, I debated and spent far too much time trying to come up with nice variable names for these, and I still wasn't happy with the result. :-)

markSpace is a bad name IMHO, as it confuses marks and spaces together. markPulse ambiguous for different reasons. It's not really a pulse.

It really needs a name "output_value_used_to_turn_on_the_LED", so I shortened it to outputOn etc etc.
I'm open to ideas, seriously, it's not trivial, but if you can come up with something that isn't confusing with the rest of the code, I'm all for it.
'on_value' & 'off_value' were other considerations but I don't think they are much of an improvement.